src/scipp/compat/pandas_compat.py

```diff
@@ -3,53 +3,108 @@
+    else:
+        values = se.to_numpy()
     return DataArray(
-        data=Variable(values=se.values, dims=[row_index_name]),
-        coords={row_index_name: Variable(dims=[row_index_name], values=row_index)},
+        data=array(values=values, dims=[row_index_name], unit=unit),
+        coords=coords,
         name=name,
     )
 
 
-def from_pandas_dataframe(df: pd.DataFrame) -> Dataset:
+def from_pandas_dataframe(
+    df: pd.DataFrame,
+    *,
+    data_columns: Optional[Union[str, Iterable[str]]] = None,
+    include_index: bool = True,
+    header_parser: HeaderParserArg = None,
+) -> Dataset:
     import pandas as pd
 
-    row_index = df.axes[0]
-    row_index_name = row_index.name or "row"
-
-    if df.ndim == 1:
-        # Special case for 1d dataframes, treat them as a series, but still
-        # wrap them in a dataset object for consistency of return types.
-        return Dataset(data={row_index_name: from_pandas_series(pd.Series(df))})
+    columns = (
+        from_pandas_series(
+            pd.Series(df[column_name]),
+            include_index=include_index,
+            header_parser=header_parser,
+        )
+        for column_name in df.axes[1]
+    )
+    coords = {da.name: da for da in columns}
 
-    sc_data = {}
-    for column_name in df.axes[1]:
-        sc_data[f"{column_name}"] = from_pandas_series(pd.Series(df[column_name]))
+    if data_columns is None:
+        data = coords
+        coords = {}
+    else:
+        if isinstance(data_columns, str):
+            data_columns = (data_columns,)
+        data = {name: coords.pop(name) for name in data_columns}
+        coords = {name: coord.data for name, coord in coords.items()}
 
-    return Dataset(data=sc_data)
+    return Dataset(data, coords=coords)
 
 
-def from_pandas(pd_obj: Union[pd.DataFrame, pd.Series]) -> VariableLike:
+def from_pandas(
+    pd_obj: Union[pd.DataFrame, pd.Series],
+    *,
+    data_columns: Optional[Union[str, Iterable[str]]] = None,
+    include_index: bool = True,
+    header_parser: HeaderParserArg = None,
+) -> VariableLike:
     """Converts a pandas.DataFrame or pandas.Series object into a
     scipp Dataset or DataArray respectively.
 
     Parameters
     ----------
     pd_obj:
         The Dataframe or Series to convert.
+    data_columns:
+        Select which columns to assign as data.
+        The rest are returned as coordinates.
+        If ``None``, all columns are assigned as data.
+        Use an empty list to assign all columns as coordinates.
+    include_index:
+        If True, the row index is included in the output as a coordinate.
+    header_parser:
+        Parses each column header to extract a name and unit for each data array.
+        By default, it returns the column name and uses the default unit.
+        Builtin parsers can be specified by name:
+
+        - ``"bracket"``: See :func:`scipp.compat.pandas_compat.parse_bracket_header`.
+
+        Before implementing a custom parser, check out
```

Comment:
Give an example here what this does (i.e., what "bracket" refers to)? (inline, not a working code)